tests/unit/test_session.c

```diff
@@ -0,0 +1,71 @@
+#include "sentry_testsupport.h"
+#include "sentry_session.h"
+#include "sentry_envelope.h"
+#include "sentry_value.h"
+#include <sentry.h>
+
+static void
+send_envelope(sentry_envelope_t *envelope, void *data)
+{
+    uint64_t *called = data;
+    *called += 1;
+
+    TEST_CHECK_INT_EQUAL(sentry__envelope_get_item_count(envelope), 1);
+
+    const sentry_envelope_item_t *item = sentry__envelope_get_item(envelope, 0);
```

Comment:
maybe also add a second test to see what happens when you manually call `capture_event`…